### ISSUE TITLE
idsAvailable NullPointerException

### ISSUE THREAD

**Issue by jvanderwee** [OneSignal/OneSignal-Android-SDK#178]
Seeing the following crash infrequently on the latest version:

```
Caused by java.lang.NullPointerException: Attempt to invoke interface method 'void com.onesignal.u$c.a(java.lang.String, java.lang.String)' on a null object reference
        at com.onesignal.OneSignal.internalFireIdsAvailableCallback(OneSignal.java:894)
        at com.onesignal.OneSignal.idsAvailable(OneSignal.java:868)
```

**Comment by jkasten2** [comment]
@jvanderwee Thanks for the crash log. From a quick review of the code it seems our SDK method isn't thread safe.
https://github.com/OneSignal/OneSignal-Android-SDK/blob/3.4.3/OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java#L882-L898

We will look into fixing it but if you can ensure you never call `idsAvailable` again until it returns with a non-null `pushToken` then you won't see this crash.

Thanks.

**Comment by jkasten2** [comment]
Fixed, will be included in the 3.4.4 release.
